src/db.ts

```diff
@@ -36,17 +38,28 @@ export class DB {
     /** Postgres connection pool for primary database access. */
     postgres: Pool
     /** Redis used for various caches. */
-    redis: Redis.Redis
+    redisPool: GenericPool<Redis.Redis>
+
     /** Kafka producer used for syncing Postgres and ClickHouse person data. */
     kafkaProducer?: Producer
     /** ClickHouse used for syncing Postgres and ClickHouse person data. */
     clickhouse?: ClickHouse
 
     constructor(postgres: Pool, redis: Redis.Redis, kafkaProducer?: Producer, clickhouse?: ClickHouse) {
         this.postgres = postgres
-        this.redis = redis
         this.kafkaProducer = kafkaProducer
         this.clickhouse = clickhouse
+        this.redisPool = createPool<Redis.Redis>(
+            {
+                create: () => Promise.resolve(redis),
+                destroy: () => Promise.resolve(undefined),
+            },
+            {
+                min: 1,
+                max: 1,
```

Comment:
With max 1 this doesn't really test pooling, how about 2?

Reply:
Not testing pooling, testing locking ;)